### ISSUE TITLE
allReady never ready since 5.0.4

### ISSUE THREAD

**Issue by jsevellec** [fluttercommunity/get_it#150]
Hi,

First of all, thanks for this really good library. It's really useful in my project.

I've got an regression since the version 5.0.4 which was ok in the 5.0.3.

Basically I'm registering a first async singleton object which is needed for 2 others async singleton objects from the same class that i'm registering with 2 different names. I'm then using then getIt.allReady to wait until all objects are available.

The problem in the 5.0.4 is that calling allready never complete.

Here is a simplified test case of my problem that shows the issue : 
the test case is working ok on 5.0.3 but fail in 5.0.4 : 

```
import 'package:get_it/get_it.dart';
import 'package:test/test.dart';

class Father {}

class Child {
  final Father father = GetIt.instance.get<Father>();
}

void main() {
  setUp(() async {
    await GetIt.I.reset();
  });

  test('async issue', () async {
    final getIt = GetIt.instance;

    getIt.registerSingletonAsync<Father>(() => Future.value(Father()));

    getIt.registerSingletonAsync<Child>(() => Future.value(Child()),
        instanceName: "childOne", dependsOn: [Father]);

    getIt.registerSingletonAsync<Child>(() => Future.value(Child()),
        instanceName: "childTwo", dependsOn: [Father]);

    await getIt.allReady(timeout: Duration(seconds: 10));

    print("never ready in 5.0.4");
  });
}
```

I've since that there's a bunch of changes in the commits between the 5.0.3 and the 5.0.4 which are related to that.

Am I doing something wrong? Is there an alternative?

Let me know if you need anything else so i can help to resolve this issue

Cheers

Jérémy

**Comment by escamoteur** [comment]
Hi Jérémy,

I just had a look at it and there is definitely a bug but that leads to an exception for me and not that allReady isn't coming back.

Can you confirm that it doesn't come back?

**Comment by jsevellec** [comment]
Hi @escamoteur 

Yes there's an exception because I've set a timeout on allReady.

If you remove the timeout on the test
```
// await getIt.allReady(timeout: Duration(seconds: 10));
    await getIt.allReady();
```

allReady never completes. And it's basically what is happening in my app

**Comment by escamoteur** [comment]
That is pretty weird because I can't reproduce that one here. There is a bug in 5.0.4 that throws a 'Father' is not registered error, which I can understand, but the not coming back I don't understand.

**Comment by escamoteur** [comment]
I just pushed my new version to github, could you try with a reference to the github repo?

**Comment by jsevellec** [comment]
Hum that's weird.

Sure thing. I'll do it right now

**Comment by jsevellec** [comment]
I've tried with your last fix from github and allReady now complete. 

It's also working in the context of my app

**Comment by escamoteur** [comment]
ah good then I will push a new version

**Comment by jsevellec** [comment]
perfect thanks!

**Comment by escamoteur** [comment]
fixed in V5.0.5
